android/src/app/organicmaps/routing/NavigationController.java

```diff
@@ -181,9 +181,12 @@ public void doBackground()
 
   private void updateVehicle(RoutingInfo info)
   {
-    SpannableStringBuilder nextTurnDistance = Utils.formatDistance(mFrame.getContext(), info.distToTurn);
-    mNextTurnDistance.setText(nextTurnDistance);
-    info.carDirection.setTurnDrawable(mNextTurnImage);
+    if (info.distToTurn.isValid())
```

Comment:
The code above makes return nullptr.

Reply:
It's `@NonNull` here

Reply:
So, seems like if null is filtered before, we can ensure here that info is always valid, no?

Reply:
You're right